cmd/daemon/kubernetes/daemonsets.go

```diff
@@ -0,0 +1,132 @@
+package kubernetes
+
+import (
+	"context"
+	"time"
+
+	"github.com/lunarway/release-manager/internal/http"
+	"github.com/lunarway/release-manager/internal/log"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+func (c *Client) HandleNewDaemonSets(ctx context.Context) error {
+	watcher, err := c.clientset.AppsV1().DaemonSets("").Watch(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			watcher.Stop()
+		case e, ok := <-watcher.ResultChan():
+			if !ok {
+				return ErrWatcherClosed
+			}
+			if e.Object == nil {
+				continue
+			}
+			if e.Type == watch.Deleted {
+				continue
+			}
+			ds, ok := e.Object.(*appsv1.DaemonSet)
+			if !ok {
+				continue
+			}
+
+			if !isDaemonSetCorrectlyAnnotated(ds) {
+				continue
+			}
+
+			// Avoid reporting on pods that has been marked for termination
+			if isDaemonSetMarkedForTermination(ds) {
+				continue
+			}
+
+			log.Infof("Observed: %d, Gen: %d", ds.Status.ObservedGeneration, ds.Generation)
+
+			// Verify if the DaemonSet fulfills the criterias for a succesful release
+			ok = isDaemonSetSuccessful(ds)
+			if !ok {
+				continue
+			}
+
+			// Check if the DaemonSet is marked with the lunarway.com/released annotation
+			if isDaemonSetMarkedAsReleased(ds) {
```

Comment:
@Crevil @emilingerslev Any ideas on how to use the annotation to detect the new deployments that are applied instead of replaced? 

This will skip updates on all new apply's as the annotation will stay. 

We could maybe use the annotation `lunarway.com/artifact-id` that we already manages in this project? If that value is changed, it's a new release?